GoChat

Simaky opened the pull request:
- rewrote server with channels
- changed slice of users to map
- fixed client

Simaky committed:
test

anastasiia-skliar commented on client/client.go:
```diff
@@ -0,0 +1,45 @@
+func sendMessage(connection net.Conn) {
+	for {
+		reader := bufio.NewReader(os.Stdin)
+		text, _ := reader.ReadString('\n')
```
error check skipped

anastasiia-skliar commented on server/server.go:
```diff
@@ -0,0 +1,93 @@
+	userName := bufio.NewScanner(connection)
+	userName.Scan()
+
+	Users[connection] = userName.Text()
```
write and read to map is not concurrent safe. 
Mutex needed

something like this
type Users struct{
    data map[string]user
    sync.Mutex
}

anastasiia-skliar commented on server/server.go:
```diff
@@ -0,0 +1,93 @@
+
+		if err != nil {
+			log.Println("User can't join to server. Error: ", err.Error())
+			return
```
return will quit all function, should all server stop because of one client ?

anastasiia-skliar commented on server/server.go:
```diff
@@ -0,0 +1,93 @@
+
+	log.Println("Server is running!")
+
+	server, err := net.Listen("tcp", ":8080")
```
move port to command line argument. 
Read about os.Args

Simaky committed:
Create .gitignore

Simaky committed:
- Err check fixed
- Mutex added
- Fixed return bug
- Server port and client address+port use os.Args
- Added Users and user structure
- Some minor bug fixes

Simaky committed:
Merge remote-tracking branch 'origin/client-server' into client-server

anastasiia-skliar commented on client/client.go:
```diff
@@ -24,15 +24,23 @@ func getMessage(connection net.Conn) {
 
-	connection, err := net.Dial("tcp", "localhost:8080")
+	argsAddress := os.Args[1]
+	argsPort := os.Args[2]
```
What will be, if there will be no Args? or less that 2 ?

Simaky committed:
- changed os.Args to flag

Simaky committed:
- Don't send your messages to yourself
